File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode insertIntoBST(TreeNode root, int val) {
        //Base Case
        if(root==null){
            TreeNode ourNode = new TreeNode(val);
            return ourNode;
        }
        
        if(val < root.val) {
            root.left = insertIntoBST(root.left,val);
        }
        else root.right = insertIntoBST(root.right,val);
        
        return root;
    }
}

File: 0701-insert-into-a-binary-search-tree/0701-insert-into-a-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        //Base Case
        if(root == nullptr) {
            TreeNode* myNewTreeNode = new TreeNode(val);
            return myNewTreeNode;
        }
        
        //search for right position
        if(val < root->val) {
            root->left = insertIntoBST(root->left,val);
        }
        else root->right = insertIntoBST(root->right,val);
        
        return root;
    }
};